txnsync/txngroups.go

```diff
@@ -17,23 +17,417 @@
+	BitmaskFee         bitmask             `codec:"feebm"`
+	FirstValid         []basics.Round      `codec:"fv,allocbound=maxEncodedTransactionGroup"`
+	BitmaskFirstValid  bitmask             `codec:"fvbm"`
+	LastValid          []basics.Round      `codec:"lv,allocbound=maxEncodedTransactionGroup"`
+	BitmaskLastValid   bitmask             `codec:"lvbm"`
+	Note               [][]byte            `codec:"note,allocbound=maxEncodedTransactionGroup"` // TODO whats the correct allocbound?
+	BitmaskNote        bitmask             `codec:"notebm"`
+	GenesisID          string              `codec:"gen"`
+	BitmaskGenesisID   bitmask             `codec:"genbm"`
+	GenesisHash        crypto.Digest       `codec:"gh"`
+	BitmaskGenesisHash bitmask             `codec:"ghbm"`
+
+	BitmaskGroup bitmask `codec:"grpbm"`
+
+	Lease        []byte  `codec:"lx,allocbound=maxAddressBytes"`
+	BitmaskLease bitmask `codec:"lxbm"`
+
+	RekeyTo        []byte  `codec:"rekey,allocbound=maxAddressBytes"`
+	BitmaskRekeyTo bitmask `codec:"rekeybm"`
+}
+
+type encodedKeyregTxnFields struct {
+	_struct                 struct{}       `codec:",omitempty,omitemptyarray"`
+	VotePK                  []byte         `codec:"votekey,allocbound=maxAddressBytes"`
+	BitmaskVotePK           bitmask        `codec:"votekeybm"`
+	SelectionPK             []byte         `codec:"selkey,allocbound=maxAddressBytes"`
+	BitmaskSelectionPK      bitmask        `codec:"selkeybm"`
+	VoteFirst               []basics.Round `codec:"votefst,allocbound=maxEncodedTransactionGroup"`
+	BitmaskVoteFirst        bitmask        `codec:"votefstbm"`
+	VoteLast                []basics.Round `codec:"votelst,allocbound=maxEncodedTransactionGroup"`
+	BitmaskVoteLast         bitmask        `codec:"votelstbm"`
+	VoteKeyDilution         []uint64       `codec:"votekd,allocbound=maxEncodedTransactionGroup"`
+	BitmaskVoteKeyDilution  bitmask        `codec:"votekdbm"`
+	Nonparticipation        []bool         `codec:"nonpart,allocbound=maxEncodedTransactionGroup"`
+	BitmaskNonparticipation bitmask        `codec:"nonpartbm"`
+}
+
+type encodedPaymentTxnFields struct {
+	_struct struct{} `codec:",omitempty,omitemptyarray"`
+
+	Receiver        []byte              `codec:"rcv,allocbound=maxAddressBytes"`
+	BitmaskReceiver bitmask             `codec:"rcvbm"`
+	Amount          []basics.MicroAlgos `codec:"amt,allocbound=maxEncodedTransactionGroup"`
+	BitmaskAmount   bitmask             `codec:"amtbm"`
+
+	CloseRemainderTo        []byte  `codec:"close,allocbound=maxAddressBytes"`
+	BitmaskCloseRemainderTo bitmask `codec:"closebm"`
+}
+
+type encodedAssetConfigTxnFields struct {
+	_struct struct{} `codec:",omitempty,omitemptyarray"`
+
+	ConfigAsset        []basics.AssetIndex `codec:"caid,allocbound=maxEncodedTransactionGroup"`
+	BitmaskConfigAsset bitmask             `codec:"caidbm"`
+
+	AssetParams        []basics.AssetParams `codec:"apar,allocbound=maxEncodedTransactionGroup"`
```

Comment:
this one is going to be pretty inefficient - we should "flatten" this one onto the `encodedAssetConfigTxnFields` so that it won't include all the child keys over and over.